extensions/ql-vscode/src/query-server/legacy/run-queries.ts

```diff
@@ -214,6 +214,27 @@ export async function clearCacheInDatabase(
   return qs.sendRequest(messages.clearCache, params, token);
 }
 
+export async function trimCacheInDatabase(
+  qs: qsClient.QueryServerClient,
+  dbItem: DatabaseItem,
+  token: CancellationToken,
+): Promise<messages.ClearCacheResult> {
+  if (dbItem.contents === undefined) {
+    throw new Error("Can't clear the cache in an invalid database.");
```

Comment:
```suggestion
    throw new Error("Can't trim the cache in an invalid database.");
```

Reply:
Yeah, I think I just have to un-commit all the `legacy/*` files from this PR, since the legacy server has been completely deprecated already.